CRM/Civicase/Hook/PreProcess/AddCaseAdminSettings.php

```diff
@@ -3,10 +3,34 @@
 use CRM_Civicase_ExtensionUtil as ExtensionUtil;
 
 /**
- * Class CRM_Civicase_Hook_PreProcess_AddCaseAdminSettings.
+ * Adds custom settings related to civicase.
  */
 class CRM_Civicase_Hook_PreProcess_AddCaseAdminSettings {
 
+  /**
+   * Single case role setting name.
+   */
+  const CIVICASE_SINGLE_CASE_ROLE_PER_TYPE = 'civicaseSingleCaseRolePerType';
+
+  /**
+   * Multiple client setting name.
+   */
+  const CIVICASE_ALLOW_MULTIPLE_CLIENTS = 'civicaseAllowMultipleClients';
+
+  /**
+   * Fetches settings from xml file.
+   *
+   * @var CRM_Case_XMLProcessor_Process
+   */
+  private $xmlProcessor;
+
+  /**
+   * Initialize dependencies.
+   */
+  public function __construct() {
```

Comment:
well in my opinion I think its nice to inject the dependencies into the class instead of class creating dependencies itself but if that is not a possibility then I think this way is a bit better as I have a reference to the dependency within my class so lets say for example if in future another method requires this xmlprocessor object so it wont need to create a new object again and creating it in constructor means my objects will always be consistent but kindly do let me know if you have some different thoughts on it

Reply:
I agree with injecting dependencies. The reason I said to remove the code now is because we don't need to instantiate this object any time the class is created. For example class will always be instantiated even if the hook does not run i.e if `shouldRun` fails. 

If any other function needs to use this later, we can instantiate the object in the run class and pass it as a parameter to whatever function needs to use it in the class.